src/main/webapp/dev/js/common/questionRubric.es6

```diff
@@ -1,15 +1,162 @@
 import {
+    showModalAlert,
     showModalConfirmation,
 } from './bootboxWrapper.es6';
 
 import {
+    ParamsNames,
     StatusType,
 } from './const.es6';
 
 import {
     disallowNonNumericEntries,
 } from './ui.es6';
 
+const EDIT_STATUS_HAS_RESPONSES = 'hasResponses';
+
+function deleteExistingResponses(questionNum) {
+    const $form = $(`#form_editquestion-${questionNum}`);
+    const COURSE_ID = $form.children(`input[name="${ParamsNames.COURSE_ID}"]`).val();
+    const FEEDBACK_SESSION_NAME = $form.children(`input[name="${ParamsNames.FEEDBACK_SESSION_NAME}"]`).val();
+    const FEEDBACK_QUESTION_ID = $form.children(`input[name="${ParamsNames.FEEDBACK_QUESTION_ID}"]`).val();
+    const USER = $form.children(`input[name="${ParamsNames.USER}"]`).val();
+    const TOKEN = $form.children(`input[name="${ParamsNames.SESSION_TOKEN}"]`).val();
+
+    $.ajax({
+        type: 'POST',
+        url: '/page/instructorFeedbackDeleteExistingResponses',
+        data: {
+            courseid: COURSE_ID,
+            fsname: FEEDBACK_SESSION_NAME,
+            questionid: FEEDBACK_QUESTION_ID,
+            user: USER,
+            token: TOKEN,
+        },
+        success(data) {
+            if (data.statusCode === '200') {
+                showModalAlert('Notice', data.message, 'OK', StatusType.SUCCESS);
+                $(`#form_editquestion-${questionNum}`).removeAttr('editstatus');
+            } else {
+                showModalAlert('Notice', data.message, 'OK', StatusType.DANGER);
+            }
+        },
+    });
+}
+
+function checkExistingResponsesBeforeMovingCols(questionNum) {
+    const EDIT_STATUS = $(`#form_editquestion-${questionNum}`).attr('editstatus');
+
+    if (EDIT_STATUS === EDIT_STATUS_HAS_RESPONSES) {
+        const DELETE_RESP_TITLE = 'Caution!';
+        const DELETE_RESP_MESSAGE_TEXT = 'This question has existing responses. Moving rubric columns requires deletion'
+                + ' of existing responses. Do you really want to delete existing responses?';
+        const DELETE_RESP_BTN_TEXT = 'Delete Responses';
+        const DELETE_RESP_CANCEL_BTN_TEXT = 'Cancel';
+        const DELETE_RESP_COLOR = StatusType.DANGER;
+
+        showModalConfirmation(
+                DELETE_RESP_TITLE,
+                DELETE_RESP_MESSAGE_TEXT,
+                () => {
+                    deleteExistingResponses(questionNum);
+                },
+                null,
+                DELETE_RESP_BTN_TEXT,
+                DELETE_RESP_CANCEL_BTN_TEXT,
+                DELETE_RESP_COLOR,
+        );
+    }
+}
+
+function swapRubricCol(questionNum, colIndex, isSwapLeft) {
```

Comment:
Taking the jQuery object of rubric edit table will only make the code more complicated. Since individual input elements have their unique IDs which can be accessed by the use of question number, the code will be much simpler if we have the question number.